### PR TITLE
370 stripe webhooks

### LINKED ISSUE #370: Add webhook handling for Stripe events
- [ ] We can do a plug maybe for deseralizing events of any type
- [ ] Controllers will handle creating/updating the right things based on the events we're getting (cards, etc)
- [ ] There needs to be two webhooks on Stripe, an account webhook and a connect webhook

I think we want to handle the following events:
- [ ] `account.updated` describes an account (on the connect webhook)

> Occurs whenever an account status or property has changed.

We will want to update the `StripeConnectAccount` with the new information.

- [ ] `customer.subscription.updated` describes a subscription (on the connect webhook)

> Occurs whenever a subscription changes. Examples would include switching from one plan to another, or switching status from trial to active.

We will want to update the `StripeConnectSubscription` with the new information, as well as update the total monthly amount that the project is currently raising.

### PR DESCRIPTION
**Pull request by begedin**
# What's in this PR?

For now, just a basic webhook handling infrastrucutre.

It handles `customer.subscription.updated` and `account.updated` on the connect webhook endpoint, but al catches all other webhooks and reponds with a 200.

The handling is basic and probably incorrect right now. 

We should discuss what exactly we need to do with it.

## References
Progress on #370

### CODE REVIEW

File: web/router.ex [hunk 1]
```diff
@@ -36,6 +36,12 @@ defmodule CodeCorps.Router do
   end
 
   scope "/", CodeCorps do
+    pipe_through :stripe_webhooks
+    post "/stripe_connect_webhooks", StripeConnectWebhooksController, :webhook
+    post "/stripe_account_webhooks", StripeAccountWebhooksController, :webhook
+  end
```
Comment (joshsmith) [inline web/router.ex]: Not sure why added these when there are some already created below, with controllers.

File: lib/code_corps/stripe_service/events/customer_subscription_updated.ex [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
-    stripe_subscription = data |> retrieve_subscription
-    local_subscription = data |> load_subscription
+    {:ok, stripe_subscription} = data |> retrieve_subscription
+    {:ok, params} = stripe_subscription |> Adapters.StripeConnectSubscription.to_params(%{})
+
+    {:ok, subscription} = data |> load_subscription |> update_subscription(params)
+    {:ok, project} = subscription |> get_project |> update_project_totals
 
-    local_subscription |> update(stripe_subscription)
+    {:ok, subscription, project}
```
Comment (joshsmith) [inline lib/code_corps/stripe_service/events/customer_subscription_updated.ex:20]: This looks like a good place to use `with`.

**Comment by joshsmith** [comment]
@begedin after a tremendous amount of back-and-forth, finally settled on there actually being a `current` for donation goals. As such, little progress made on the actual webhooks portion of this PR.

File: lib/code_corps/donation_goals_manager.ex [hunk 3]
```diff
@@ -49,26 +49,49 @@ defmodule CodeCorps.DonationGoalsManager do
...
   end
 
-  def set_current_goal_for_project(%Project{} = project) do
-    project
-    |> find_current_goal
-    |> set_to_current(project)
+  def update_related_goals(%{donation_goal: %DonationGoal{} = donation_goal}), do: update_related_goals(donation_goal)
+  def update_related_goals(%DonationGoal{project_id: project_id}) do
+    with project <- Repo.get(Project, project_id),
+         current_goal <- find_current_goal(project)
```
Comment (begedin) [inline lib/code_corps/donation_goals_manager.ex:95]: Indent

File: lib/code_corps/donation_goals_manager.ex [hunk 4]
```diff
@@ -80,36 +103,14 @@ defmodule CodeCorps.DonationGoalsManager do
...
-
-  defp find_largest_goal(%Project{id: project_id}) do
+  defp update_goals(%Project{id: project_id} = project, %DonationGoal{} = donation_goal) do
     DonationGoal
     |> where([d], d.project_id == ^project_id)
-    |> order_by(desc: :amount)
-    |> limit(1)
-    |> Repo.one
-  end
+    |> where([d], d.id != ^donation_goal.id)
```
Comment (begedin) [inline lib/code_corps/donation_goals_manager.ex:151]: Nice one!

**Review by begedin — COMMENTED** [review]
What' I'm seeing looks good.

**Review by joshsmith — APPROVED** [review]
Approved with understand that we're not done yet and need some smaller, more targeted PRs.

**Pull request merged**